src/creator.rs

```diff
@@ -5,10 +5,9 @@ use crate::{
     units::{ControlHash, PreUnit, Unit},
     Hasher, Receiver, Round, Sender,
 };
-use futures::{channel::oneshot, FutureExt, StreamExt};
-use futures_timer::Delay;
+use futures::{channel::oneshot, StreamExt};
 use log::{info, trace, warn};
-use std::time::Duration;
+use tokio::time::{self, Duration};
 
 /// A process responsible for creating new units. It receives all the units added locally to the Dag
 /// via the parents_rx channel endpoint. It creates units according to an internal strategy respecting
```

Comment:
Can you add a comment that a unit from round 0 has no parents? and the below condition only apply when `r >= 1`?

Reply:
Also, can you add a comment that it is required that a unit can appear in the channel only after all its parents have appeared there?

Reply:
Done